From ca73befbb81886fbe27d16873bdc1407487630d7 Mon Sep 17 00:00:00 2001
From: Christian Svensson <blue@cmd.nu>
Date: Tue, 11 Jul 2023 12:03:39 +0200
Subject: [PATCH] sonix: fix mfd Makefile for N821 CPLD core module

---
 drivers/mfd/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/mfd/Makefile b/drivers/mfd/Makefile
index 4efac062143d..c70fa18892e2 100644
--- a/drivers/mfd/Makefile
+++ b/drivers/mfd/Makefile
@@ -280,4 +280,3 @@ obj-$(CONFIG_MFD_RSMU_I2C)	+= rsmu-i2c.o
 obj-$(CONFIG_MFD_RSMU_SPI)	+= rsmu-spi.o
 
 obj-$(CONFIG_CISCO_N821_CPLD)  += cisco-n821-cpld.o
-cisco-n821-cpld-objs  := cpld_core.o
-- 
2.39.3

