From 0b70210273ed00a9e8ef6b545de3c2dcf4c23854 Mon Sep 17 00:00:00 2001
From: Christian Svensson <blue@cmd.nu>
Date: Tue, 11 Jul 2023 12:14:35 +0200
Subject: [PATCH] sonix: fix N821 header layout and constant naming

---
 include/linux/mfd/cisco-n821-cpld.h | 43 ++++++++++++++---------------
 1 file changed, 21 insertions(+), 22 deletions(-)

diff --git a/include/linux/mfd/cisco-n821-cpld.h b/include/linux/mfd/cisco-n821-cpld.h
index eb8fa490e694..20ee77b33736 100644
--- a/include/linux/mfd/cisco-n821-cpld.h
+++ b/include/linux/mfd/cisco-n821-cpld.h
@@ -1,37 +1,36 @@
-// SPDX-License-Identifier: GPL-2.0
+/* SPDX-License-Identifier: GPL-2.0 */
 /*
- * cpld.c - Interface Masters CPLD module for Niagara 821
- * CPLD driver for Niagara 821 board used in Cisco vEdge 1000.
+ * Declarations for the CPLD used in the Cisco vEdge 1000 board Niagara 821
  *
  * Author: Christian Svensson <blue@cmd.nu>
  */
 
-#ifndef __HEADER__CPLD__
-#define __HEADER__CPLD__
+#ifndef __MFD_CISCO_N821_CPLD__
+#define __MFD_CISCO_N821_CPLD__
 
 #include <linux/regmap.h>
 
-#define CISCO_N821_CPLD_INT_IRQ_I2C       BIT(3)
-#define CISCO_N821_CPLD_INT_IRQ_RESET_BTN BIT(6)
-#define CISCO_N821_CPLD_INT_IRQ_OVERTEMP  BIT(7)
+#define N821_CPLD_INT_IRQ_I2C       BIT(3)
+#define N821_CPLD_INT_IRQ_RESET_BTN BIT(6)
+#define N821_CPLD_INT_IRQ_OVERTEMP  BIT(7)
 
-#define CISCO_N821_CPLD_REG_REVISION    0x2
-#define CISCO_N821_CPLD_REG_PWR_CTRL    0x3
-#define CISCO_N821_CPLD_REG_USB3_RST    0x4
-#define CISCO_N821_CPLD_REG_PWR_STATUS  0x6
-#define CISCO_N821_CPLD_REG_FAN0_SPEED  0x7
-#define CISCO_N821_CPLD_REG_FAN1_SPEED  0x8
-#define CISCO_N821_CPLD_REG_SFP_PRSNT   0x0B
-#define CISCO_N821_CPLD_REG_IRQ_STATUS  0x10
-#define CISCO_N821_CPLD_REG_IRQ_MASK    0x11
-#define CISCO_N821_CPLD_REG_PWR_TEST    0x20
+#define N821_CPLD_REG_REVISION    0x2
+#define N821_CPLD_REG_PWR_CTRL    0x3
+#define N821_CPLD_REG_USB3_RST    0x4
+#define N821_CPLD_REG_PWR_STATUS  0x6
+#define N821_CPLD_REG_FAN0_SPEED  0x7
+#define N821_CPLD_REG_FAN1_SPEED  0x8
+#define N821_CPLD_REG_SFP_PRSNT   0x0B
+#define N821_CPLD_REG_IRQ_STATUS  0x10
+#define N821_CPLD_REG_IRQ_MASK    0x11
+#define N821_CPLD_REG_PWR_TEST    0x20
 
-#define CISCO_N821_CPLD_I2C_REG_CTRL    0x2
+#define N821_CPLD_I2C_REG_CTRL    0x2
 
 enum cisco_n821_cpld_irqs {
-  CISCO_N821_CPLD_IRQ_I2C,
-  CISCO_N821_CPLD_IRQ_RESET_BTN,
-  CISCO_N821_CPLD_IRQ_OVERTEMP,
+  N821_CPLD_IRQ_I2C,
+  N821_CPLD_IRQ_RESET_BTN,
+  N821_CPLD_IRQ_OVERTEMP,
 };
 
 #endif
-- 
2.39.3

