From cad6591a73652ad309504b291c761d132283096a Mon Sep 17 00:00:00 2001
From: Christian Svensson <blue@cmd.nu>
Date: Thu, 22 Jun 2023 13:40:01 +0200
Subject: [PATCH] sonix: wait for regmap to be available for n821 cpld

---
 drivers/staging/n821-cpld/cpld_core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/n821-cpld/cpld_core.c b/drivers/staging/n821-cpld/cpld_core.c
index 81c6372244e9..2cf4c7fe8459 100644
--- a/drivers/staging/n821-cpld/cpld_core.c
+++ b/drivers/staging/n821-cpld/cpld_core.c
@@ -88,8 +88,8 @@ static int n821_cpld_probe(struct platform_device *pdev)
 
   regmap = syscon_node_to_regmap(dev->of_node);
   if (IS_ERR(regmap)) {
-    dev_err(dev, "cannot get regmap\n");
-    return PTR_ERR(regmap);
+    dev_info(dev, "cannot get regmap, try again later\n");
+    return -EPROBE_DEFER;
   }
 
   irq = irq_of_parse_and_map(dev->of_node, 0);
-- 
2.39.3

